### PR TITLE
Add ZodString Datetime support

### PR DESCRIPTION
**Pull request by samchungy**
Heya 🚤 

Since I added this bad boy to Zod, I figured I should be the one to implement it here. Not sure how we want to fix the type support here.

We could extend Zod in this library or we could just bump up the dev deps as this code should work with both versions. Happy with either, just let me know 😄

### CODE REVIEW

File: src/openapi-generator.ts [hunk 1]
```diff
@@ -587,6 +587,10 @@ export class OpenAPIGenerator {
       return 'uri';
     }
 
+    if (zodString.isDatetime?.()) {
```
Comment (samchungy) [inline src/openapi-generator.ts:590]: Optional chaining will mean backwards compatibility

**Comment by AGalabov** [comment]
@samchungy sorry ... I somehow missed that there is a PR open among all other issue discussions 🤦. I'll have a look in a bit

File: src/openapi-generator.ts [hunk 2]
```diff
@@ -587,6 +587,10 @@ export class OpenAPIGenerator {
       return 'uri';
     }
 
+    if (zodString.isDatetime) {
```
Comment (AGalabov) [inline src/openapi-generator.ts:590]: I love when I see that it's as easy as that to add a new feature :bow:

**Review by AGalabov — APPROVED** [review]

**Comment by AGalabov** [comment]
@samchungy I am okay with keeping the peer dependencies untouched and just bumping the dev dependencies for now. I'll be merging the PR and scheduling a release in a bit (once again :smile:). Thank you!

**Pull request merged**